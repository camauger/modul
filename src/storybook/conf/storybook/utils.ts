export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';

