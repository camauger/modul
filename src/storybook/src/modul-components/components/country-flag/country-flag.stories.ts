import { select, text } from '@storybook/addon-knobs';
import { COUNTRY_FLAG_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MCountryCodeISO2 } from '@ulaval/modul-components/dist/utils/country/country';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${COUNTRY_FLAG_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    props: {
        codeIso2: {
            default: select(
                'Prop code-iso2',
                Enums.toValueArray(MCountryCodeISO2),
                MCountryCodeISO2.Unitedstates
            )
        },
        width: {
            default: text(
                'Prop width',
                '4em'
            )
        },
        height: {
            default: text(
                'Prop height',
                '4em'
            )
        }
    },
    template: `<${COUNTRY_FLAG_NAME}
        :code-iso2="codeIso2"
        :width="width"
        :height="height"
    />`
});

defaultStory.story = {
    name: 'default'
};

// export const disabled = () => '<m-button :disabled="true">A Button</m-button>';
