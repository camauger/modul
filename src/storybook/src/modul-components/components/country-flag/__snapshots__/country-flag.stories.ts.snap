// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-country-flag default 1`] = `<!---->`;
